paasta_tools/kubernetes_tools.py

```diff
@@ -2552,6 +2553,19 @@ def get_topology_spread_constraints(
             "topology_spread_constraints", default_pod_topology_spread_constraints
         )
 
+    def get_projected_sa_volumes(self) -> List[ProjectedSAVolume]:
+        config_volumes = super().get_projected_sa_volumes()
+        token_config = (
+            load_system_paasta_config().get_service_auth_token_volume_config()
+        )
+        if (
+            self.service in get_authenticating_services(self.soa_dir)
+            and token_config
+            and not any(volume == token_config for volume in config_volumes)
+        ):
```

Comment:
when can this happen?

Reply:
Do you mean the `any(...)` check? Just making sure we are not trying to mount 2 equal volumes if someone manually added the SA volume setting in a service instance config.

Reply:
@piax93 oh, i just wasn't using all my neurons - i was trying to figure out how someone could write an authn.yaml that contains the token config... but authn.yaml and the projected_sa_volume config are two different things!

Reply:
(i.e., you don't need to explicitly create a `projected_sa_volume` config for the authn stuff to work)